flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/datastream/DataStreamUserDefinedFunctionITCase.scala

```diff
@@ -230,6 +231,26 @@ class DataStreamUserDefinedFunctionITCase extends StreamingMultipleProgramsTestB
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
 
+
+  @Test
+  def testScalarFunctionDeterministic(): Unit = {
```

Comment:
Yes, agree with @wuchong , Add a test in `ExpressionReductionTest` is correct. and I think it's better to add a exception test case.